flink-libraries/flink-table/src/test/scala/org/apache/flink/table/functions/aggfunctions/AggFunctionTestBase.scala

```diff
@@ -36,14 +36,23 @@ abstract class AggFunctionTestBase[T] {
 
   def aggregator: AggregateFunction[T]
 
+  def ifSupportRetraction: Boolean = true
+
   @Test
-  // test aggregate functions without partial merge
-  def testAggregateWithoutMerge(): Unit = {
+  // test aggregate and retract functions without partial merge
+  def testAccumulateAndRetractWithoutMerge(): Unit = {
     // iterate over input sets
     for ((vals, expected) <- inputValueSets.zip(expectedResults)) {
-      val accumulator = aggregateVals(vals)
-      val result = aggregator.getValue(accumulator)
+      val accumulator = accumulateVals(vals)
+      var result = aggregator.getValue(accumulator)
       validateResult(expected, result)
+
+      if (ifSupportRetraction) {
+        retractVals(accumulator, vals)
```

Comment:
If we want that `accumulate` and `merge` can be used after `merge` was applied, we should add tests for that as well.